src/nano_contracts/utils.ts

```diff
@@ -41,15 +44,16 @@ export function getContainerInternalType(
   const match = type.match(/^(.*?)\[(.*)\]/);
   const containerType = match ? match[1] : null;
   const internalType = match ? match[2] : null;
-  if ((!internalType) || (!containerType)) {
+  if (!internalType || !containerType) {
     throw new Error('Unable to extract type');
   }
   // Only some values are allowed for containerType
   switch (containerType) {
     case 'Tuple':
     case 'SignedData':
     case 'RawSignedData':
-      return [containerType, internalType]
+    case 'Optional':
+      return [containerType, internalType];
```

Comment:
Why do you need this? You already handle the Optional in the beginning of the method.

Reply:
This is something I was looking forward to ask, there are a lot of ways to make a value optional.

```py
class Foo(Blueprint):
  @public
  def foo(self, ctx: Context, val: Union[str, None]):
      noop
  @public
  def bar(self, ctx: Context, val: Optional[str]):
      noop
  @public
  def baz(self, ctx: Context, val: str | None):
      noop
```

Can we be sure all of these will evaluate to `str?` on the API?

Reply:
This seems to me an important suite of tests to be implemented outside of this PR, which is already pretty large.

That way we can optimize as needed in the future, certain that there'll be no breaking changes.

Reply:
We already have tests for all Optional fields (`str?`, `int?`, `VarInt?`, etc.) and SignedData fields.
We will implement other container types and expand these tests, but as you said this is not in the scope of this PR.